### ISSUE TITLE
Tool Selection and Strategy

### ISSUE THREAD

**Issue by Chrissuli** [elmsln/elmsln#514]
Hi, sorry to put this here on github but, I been having a bit of difficulty finding a suitable contact point for finding out more about ELMS-LN.

We have been looking for a suitable Edu  platform that supports or allows us to develop an integrated platform (LMS, LRS / CMS, Ecommerce & web services etc.).  The convergence of the markets has made Drupal based systems a clear favourite but, it is a large and rather scary beast for a small edu startup like ourselves. So far we seem to have a shortlist of Connect i platform, Opigno or ELMS-LN. I have also evaluated moodle and in paricular Sakai, from which I derived a lot of good ideas and requirements. Opigno looks simple enough to getvstarted, but then threatens a very long and steep learning curve for the future, not to mention costs for further development and customization.

These include the ability to structure modules and lessons in a multi-level  /lattice structure with workflow controls or student  open choice using learning roadmaps, 360 degree feedback and review processes and controls to facilitate self study and regulation as much as possible and the ability for students to create their own student pages and infobases individually or in groups  to supplement the official lessons or content components.  

None of the traditional LM systems seem to have the capabilities needed to support the pedagogical model and learning processes I have envisioned and mapped out. It seems to me that to do this, keep up with the times and get best of breed Drupal is the way to go. However, I am not sure we (1 techie and 1 teacher / lecturer at present) have the capbilities or resources to cope with Drupal and even Opigno by ourselves. Obviously crowd and open sourcing are key aspects of our strategy for realising our dream edu-ecosystem, but small manageable steps are the only way to get started, idealy with an established platform with which we can grow and develop without having to reinvent the wheel.

My colleague however, feels that ELMS-LN is too small a development group that branched off at Drupal 6 already, and is therefore too risky both in support and continued development capabilities. I also have concerns in terms of the level of instutionalisation in the design of ELMSLN, as am more focused on an edubusiness with integrated eCommerce and web services. I would like the the functionality of ELMS-LN and the future development capabilities of Opigno/Drupal but, not the scary appearance and linear thinking that often tends to be inherent in those kinds of edu organisations, an impression that was very strong with Sakai despite its ready made functionality. 

If anyone can give some answeres to these issues and  concerns (with evidence ;) I would very much appreciate it. 

Many thanks in advance,
Chris

Education and Training Specialist
Www.spensort.com
Helsinki, Finland

**Comment by btopro** [comment]
Chris -

Thanks for reaching out. There's lots of documentation, video, presentations, blog posts, contact info and more on http://elmsln.org for future reference. we're also @ elmsln on twitter. To clear up an initial point, this is not Drupal 6. This is Drupal 7, packaged in a more sustainable way. We're using Drupal as a development framework, not a traditional site.

To your other question...
You're assessment that Learning Management systems are built on a flawed premise is one that we share, it's why this project exists (among other things). It's also why, while there are multiple developers at Penn State working on this full time with contributions at the community at large, the university still is adopting Canvas.

(mounts soapbox)
Legacy deployment methodologies have existed for years and will continue to exist over the next decade for so as automation can only stomp out faulty thinking so quickly. There will always be a need for a watered down, "traditional", "instutionalisation", factory model of education. It is our (industry our) legacy which hundreds of years built. We are the institution and you will come to us for our knowledge as we are the knowledge brokers with the haves (with knowledge) teaching the have nots of knowledge (all those paying to come to us).

So, we're in the same boat. We want to build something that's untraditional because traditional pathways to learning and the structuring of education are a dying breed. The problem with status quo is that it works great (in all situations) right up until it's jumped the shark. Status quo will never realize it needs changed until it's too late and lost relevancy.

We are here to save education from watering itself down by unleashing pedagogy and tallent from within institutions instead of outsourcing it solely to the whim of SilCo and others that have no stake in the game. We are not just some vertical on a spreadsheet in a boardroom, we are the hearts and minds of the next generation.
{gets off soapbox}

This is part of why we're here. We're also here because we feel that we are democratizing integrations, scalability, SaaS, (eventually PaaS), and everything else that currently is a barrier from a 1-person army changing the educational landscape. A system that can shift and be whatever people need it to be, powered by automation and following a solid pattern of development, can overturn tables and make technology obey pedagogy, not the other way around.

ELMSLN is a network of systems but it's also helping to create a level of "containerization" of courses while allowing developers and instructional technologists to sustain innovation. By containerizing (in our definition here, each part of each course is a Drupal site at a technical level) we're able to allow instructor A branch off in a completely different direction pedagogically from instructor B, while still keeping the underpinning maintainable from a development and code sustainability perspective.

If you plan on going the Drupal route, it's cause of integrations and that's part of where our heart is. I'd recommend you at least ping @rickhumphries who is developing against ELMSLN because of it's integrations and mindset of integration of everything. They have integration with Learning Locker to do XAPI statement tracking to issue badges upon completion of items for the concept of micro-credentialing in medical training.

If you are concerned about institutional ways of thinking I assure you we are hear to automate those boring, dead design patterns so that we can allow new forms of pedagogy to shine on top. We'll be hopefully releasing some videos of an experiment we have as far as using the studio and HTML5 webcam recording made by @kat-wehr and @curiosity26 for a business speaking which works with our Studio tool so that people can then go in and critique each other after the fact. This is video shows a course user experience we've constructed for a graphic novels course so that students can write in a self reflective journal, submit work, and critique others https://www.youtube.com/watch?v=Xf8q3vq6rgE

These examples are currently linear for sure, so your adaptive learning connotations would definitely be of interest to lead UX / faculty @michael-collins who is helping us build a program on the platform (and the platform via the program) in order to allow students to branch off and accomplish things needed.

If there's anything to be concerned with, it's Drupal in general. If you are comfortable with Drupal, then maybe this is right for you. The best way to find out is to play with the system on your local computer via Vagrant (like we are right now as we develop the system further in a coffeeshop). Our Docs section can help with that -- http://docs.elmsln.org/en/latest/development/Vagrant-Step-by-Step-setup/

If you have any more questions please feel free to reach out and if we need something more synchronous shoot us a tweet and we'll DM.

**Comment by Chrissuli** [comment]
Hi,
Thanks for the prompt reply and all the further info refernces. Dont know what time it is in your part of the world but hope you did not have to rise to early or stay late to reply.

Currently we do have a test installment of Opigno and I am working my way through the basic manual to familiarise myself with it more thoroughly. Now just have to convince my colleague that a closer look at ELMS-LN  is waranted.

Don't worry about the soap box - I meant no offense but, although a qualified teacher & HE lecturer in my own field, systems designed for HE institutions such as Sakai have a look and feel that is not suitable for the subscribers and purposes I have in mind. As you clearly recognised, flexibility, customizability and scalability without the need for an educational specialist to get stuck too deeply under the hood are key features for any future learning management systems. This is one reason why the  Drupal /Opigno concept of a Framework LMS with a modular app tools shop is appealing even if the edu expert input in the design and devlopment process seems somewhat lacking at first sight. However, I am not sure I fully follow your comment about the Canvas LMS. Do they prefer it, or is it  temporary solution till ELMSLN is ready? 

I will certainly look into the info and references you kindly provided and once I have had time to process it all, I may well be back in touch with more questions :)

Again, many thanks for your you help.

Kr,
Chris

**Comment by btopro** [comment]
I have no issue with Opigno, it's Drupal with a Moodle-flare but at the end of the day it's at least Drupal so it's more flexible. We are a different kind of monster as we're about ultra flexibility and the ability to bend expectations of what is possible (because it's all just web technology :) ).

Comment about Canvas  -- It has a certain way of working and structuring course material. It's the same routinized path of yester-year but does do some things well or at least in a more complete form then we currently do. We're all about options here so those that want to break out of normal models and have control over the learning experience and push the envelope have us as an outlet.

**Comment by btopro** [comment]
- [ ] Create a page in the documentation comparing traditional LMSs with ELMSLN and how the methodology is different
- [ ] Page speaking to relationship with Drupal, ensure we talk about version of core / multiple cores, etc

**Comment by kfurst** [comment]
(To throw in my two cents here)...

Chris, 

I administer the Moodle installation in our 2000 member organization, have worked with several other LMS systems in the past and spent a fare amount of time messing around with Opigno. I think it all really comes down to what you hope to achieve with your organization. Opigno is pretty easy to set up and get rolling out of the box, yet it suffers from the same limiting issues as other LMSs in that one quickly loses sight of andragogical (or in my case with secondary schools, pedagogical) best practices. Instead of looking to deliver instruction in a manner in which people (or in my case, cynical, jaded teenagers) learn best, we quickly focus on theme tweaks, SSO integration, and whether a unit module's assessment bank is working. We completely bypass the "learning" portion to focus solely on logistics and aesthetics. 

Further, there's something to be send from constantly developing behind the 8 ball. I spend a great deal of my day trying to find workarounds for integrating instructor's content and tools with our LMS. The trouble is if the approach and underlying structure of the LMS is outdated, it becomes this reactive death-spiral, where the workaround takes too long to develop, has too many steps, is too hard to support beyond 6 months, and is frankly too ugly for anyone to use. This project is really the first to be "out front" even if the trade off is that it isn't as "LMS in a box" as Opigno.  

Why does this matter from a business standpoint? Well, I would argue that if @btopro 's statement in one of his videos that "... the UX in all LMSs suck" is correct (which it is), it is more because of the approach than the tool. LMSs are generally limited in terms of experience because they take a unified and antiquated approach to learning which results in generally unidirectional content delivery or patronizing bidirectional content that relies on flash-based puzzles and games masquerading as learning activities. Unfortunately this is the limit if one is stuck within the confines of an LMS where all of the active  learning must take place inside a finite box (i.e. the LMS) --which isn't how real learning works. So your learners aren't going to be actively engaged but will comply with requirements if they're required to. That doesn't really bring anything new or inviting to the market. 

So to summarize... Canvas, Moodle, and even Opigno are all good tools for digitizing the standard classroom experience. They are easy to set up, relatively easy to maintain, and provide mass delivery of content using the standardized model. You can get up and running tomorrow; ELMS isn't quite to that point. Yet, those systems actively work against you (trust me, I've got grey hairs growing) if you try to approach education differently in the distributed, democratized, and truly open approach about which btopro writes. 

Thus it depends entirely on your organization's mission, values, approach and timetable.

**Comment by Chrissuli** [comment]
Hi Kevin,
Thanks for your 2 cents of input, it all counts :)

The questions you raise are highly pertinent and have been swirling around in my mind for sometime.
I am indeed trying  to approach education differently in a more distributed manner than can adpt and keep up with new techNologies. I have already mapped out both the business model and pedagogical / androgocical model / learning processes I want to use, but the difficulty as always is resources and capability. 

It would be good to get something up and running soon, but future development would undoubtedly be a lonGterm and on going process. My colleague wants something simple to start, but with the flexibility to develop, and is asking for a proof of concept on Opigno. I have been leaning more towards ELMS, as I understood from a previous recommendation that this was more than just a framework system for which extra  'plug and pray' app tools are available on a technology lead Scrum development basis.

The current problem is in establishing the facts about the systems in respect to their capability and development requirements compared to our own resources and capabilities. There seems to be some differences of understanding as to what Opigno is and can do, and I am as yet unclear about what ELMS actually provides being designed as a Learning Network rather than an LMS.

Being a couple of otherwise unemployed fellas trying to break out of the rat trap and achieve our dreams still, expensive systems and lots of development and consultancy support are not an option, now or in the near future. This is a major reason why crowd and open sourcing are key to our development strategy, along with perhaps some strategic alliances. Despite its potential, I still have the feelig with Opigno that we would be rather isolated in our efforts. Whilst Scalability and Functionality are important for the medium to long term success, in the short-term immediate situation,  Installability, Reliability, and SUPPORT are the key factors. In this respect it has been interesting to observe the differences in level and time of response in the user support forums of Opigno compared to the Github for ELMS. So far ELMS is winning hands down :)

Once I have had a chance to dig under the hood a little more with btopro's references, it would be nice to chat with you again sometime  and see what we have learned ;)

Many thanks for your insights, it is most welcome.

Kr, 
Chris

**Comment by btopro** [comment]
> I am indeed trying to approach education differently in a more distributed manner than can adpt and keep up with new technologies.

This is a large part of why the system has been set in motion the way it is. Most basic of concepts; take the dozen or so capabilities of a traditional LMS, break them out into their own smaller buckets of capabilities, improve each one independently / allow people to break from the norm / build upon conventions in different directions.

You're asking yourself great questions and exploring within your constraints as we all have who have arrived here. I'm much more the live and let live mentality and have steered many people away from our platform when questioning suggested it didn't meet their needs. I only say this cause I'm not a sales person (no one is here), we're a community seeking members who realize want need something different and want to help bring that to life :)

Anyway, please feel free to continue asking questions about the platform. We're starting to get more instructional designers / faculty active in our issue queues to help us realize what needs documented, improved, or rethought.  Vagrant is probably your best option for seeing capabilities unless you want to watch some videos. It's the easiest to get up and going, it's a VM so it's disposable, free, and lets you see the state of the project / start to get an idea for what's possible.

**Comment by btopro** [comment]
Closing out; Hope we helped, if you have any other questions please feel free to ask. Happy Educating!
